src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -369,8 +373,18 @@ private static bool TryParseEnumCore(ref Utf8JsonReader reader, JsonSerializerOp
             int charsWritten = reader.CopyString(charBuffer);
             ReadOnlySpan<char> source = charBuffer.Slice(0, charsWritten);
 
-            // Try parsing case sensitive first
-            bool success = Enum.TryParse(source, out T result) || Enum.TryParse(source, ignoreCase: true, out result);
+            bool success;
+            T result;
+            if ((_converterOptions & EnumConverterOptions.AllowNumbers) != 0 || !Regex.IsMatch(source, NumericPattern))
+            {
+                // Try parsing case sensitive first
+                success = Enum.TryParse(source, out result) || Enum.TryParse(source, ignoreCase: true, out result);
```

Comment:
This predates your changes, but I'm kind of puzzled as to why we do this. Couldn't this just have been a single parse call with `ignoreCase` set to true?

Reply:
I just guess it is for performance. Considering more common case is `source` is exactly same as Enum name, and case-sensitive comparing of `Enum.TryParse` may be faster than obscure comparing ?
Also there is STJ doc [sample](https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/converters-how-to?pivots=dotnet-6-0#sample-factory-pattern-converter) with same code pattern and the comment said 

> For performance, parse with ignoreCase:false first.

Reply:
Interesting, thanks. I'll see if I can validate that claim with a benchmark :-)

Reply:
```C#
BenchmarkRunner.Run<Benchmark>(args: args);

public class Benchmark
{
    [Params("GetProperty", "getproperty")]
    public string EnumStringValue;

    [Benchmark]
    public BindingFlags CaseInsensitive()
    {
        _ = Enum.TryParse(EnumStringValue, ignoreCase: true, out BindingFlags result);
        return result;
    }

    [Benchmark]
    public BindingFlags CaseSensitiveThenCaseInsensitive()
    {
        _ = Enum.TryParse(EnumStringValue, out BindingFlags result) || Enum.TryParse(EnumStringValue, ignoreCase: true, out result);
        return result;
    }
}
```

Gives

| Method                           | EnumStringValue | Mean     | Error    | StdDev   |
|--------------------------------- |---------------- |---------:|---------:|---------:|
| CaseInsensitive                  | getproperty     | 40.44 ns | 0.460 ns | 0.384 ns |
| CaseSensitiveThenCaseInsensitive | getproperty     | 80.20 ns | 1.615 ns | 1.586 ns |
| CaseInsensitive                  | GetProperty     | 38.47 ns | 0.636 ns | 0.595 ns |
| CaseSensitiveThenCaseInsensitive | GetProperty     | 35.09 ns | 0.506 ns | 0.473 ns |

While it's true that the current approach is marginally faster, it drastically regresses performance in case insensitive cases. I think we should just use a single method call. cc @stephentoub